shm/Ship.hpp

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include <string>
+
+#include "Cargo.hpp"
+
+class Ship
+{
+public:
+    Ship();
+    Ship(int capacity, int maxCrew, int speed, const std::string& name, size_t id);
+    Ship(int maxCrew, int speed, size_t id);
+
+    void setName(const std::string& name);
+
+    Ship& operator-=(size_t num);
+    Ship& operator+=(size_t num);
+
+    [[nodiscard]] size_t getCapacity() const { return capacity_; }
+    [[nodiscard]] size_t getMaxCrew() const { return maxCrew_; }
+    [[nodiscard]] size_t getSpeed() const { return speed_; }
+    [[nodiscard]] std::string getName() const { return name_; }
+    [[nodiscard]] size_t getId() const { return id_; }
```

Comment:
One times you have getter in .hpp other time in .cpp. Please chose one version and apply in whole project.